framer/TextLayer.coffee

```diff
@@ -93,15 +93,27 @@ class exports.TextLayer extends Layer
 		super options
 		@__constructor = true
 
-		# Keeps track if the width or height are explicitly set, so we shouldn't update it afterwards
-		if not options.autoSize? and not options.truncate
+		# the goal is
+		# autoSize elements should not soft wrap, only hard wrap (based on newlines)
+		# fixed (given a box) elements should soft wrap
+
+		if options.autoSize
+			@autoWidth = true
+			@autoHeight = true
+		else if options.autoSize isnt false and not options.truncate
+			# if not explicitly disabled auto sizing, auto size width/height, unless they were explicitly set
 			if not options.autoWidth?
-				explicitWidth = options.width? or _.isNumber(options?.size) or options?.size?.width? or options?.frame?.width?
+				explicitWidth = options.width? or _.isNumber(options.size) or options.size?.width? or options.frame?.width?
 				@autoWidth = not explicitWidth
 			if not options.autoHeight?
-				explicitHeight = options.height? or _.isNumber(options?.size) or options?.size?.height? or options?.frame?.height?
+				explicitHeight = options.height? or _.isNumber(options.size) or options.size?.height? or options.frame?.height?
```

Comment:
Why dit you remove the optionality of the `options` here?

Reply:
`options` is never null